### PR TITLE
feat(data-plane/pulumi): Update nile-js to 0.17.0 with authToken support; update connect()

### PR DESCRIPTION
**Pull request by ybyzek**
This PR makes the following enhancements:

- Update nile-js to v0.17.0
- Sets up support for authToken
- Uses the new `connect` API

### CODE REVIEW

**Review by scottydawg — APPROVED** [review]

File: data-plane/pulumi/src/commands/reconcile/index.ts [hunk 1]
```diff
@@ -35,10 +25,21 @@ export default class Reconcile extends Command {
       workspace,
       email,
       password,
+      authToken,
     } = flags;
 
     // nile setup
-    await this.connectNile({ basePath, workspace, email, password });
+    this.nile = await NileApi.connect(
```
Comment (jrea) [inline data-plane/pulumi/src/commands/reconcile/index.ts:32]: this interface is not correct.

Comment (jrea) [reply data-plane/pulumi/src/commands/reconcile/index.ts:32]: https://github.com/TheNileDev/nile-js/commit/09f809b102ff6a3c44d9a5620c7d88caa4be6841#r82490168

**Comment by ybyzek** [comment]
Converting PR to `Draft` until https://github.com/TheNileDev/examples/pull/16#discussion_r957349618 is resolved (and example is updated as needed)

File: data-plane/pulumi/src/commands/reconcile/index.ts (hunk 1 above)
Comment (ybyzek) [reply data-plane/pulumi/src/commands/reconcile/index.ts:32]: @jrea is the interface change going to result in impact to how it's called here in the example?  Is this example waiting on v0.16.3?

**Comment by ybyzek** [comment]
Update: this PR will wait for v0.16.3 because the `connect()` interface will change

**Comment by ybyzek** [comment]
@jrea @scottydawg  latest commit updated the JS client to 0.17.0, now using the updated `connect()` interface.

**Review by scottydawg — APPROVED** [review]

**Review by jrea — APPROVED** [review]

**Pull request merged**